substratevm/src/com.oracle.graal.pointsto/src/com/oracle/graal/pointsto/AnalysisPolicy.java

```diff
@@ -143,13 +143,13 @@ public abstract AbstractSpecialInvokeTypeFlow createSpecialInvokeTypeFlow(Byteco
                     TypeFlow<?>[] actualParameters, ActualReturnTypeFlow actualReturn, BytecodeLocation location);
 
     @SuppressWarnings("unused")
-    public int makePoperties(BigBang bb, AnalysisObject... objects) {
+    public int makeProperties(StaticAnalysisEngine analysis, AnalysisObject... objects) {
```

Comment:
I don't like that variables/parameters/fields are now called `analysis` instead of `bb`. `analysis` is a very generic term. But `staticAnalysisEngine` everywhere is of course annoyingly long.

So why not do the renaming "the other way round" and keep the name `BigBang` / variable name `bb` for what you now call `StaticAnalysisEngine`? The implementation classes can then be properly named `ReachabilityAnalysis` and `PointsToAnalysis` - but there are very few cases where you actually need to cast `BigBang` to `PointsToAnalysis`.

To minimize renamings, you can also keep the variable name `bb` when the type is actually already cast to `PointsToAnalysis`.

Reply:
I was feeling that the shift from `BigBang` to `StaticAnalysisEngine` was a good one because the `BigBang/Inflation` seemed more confusing than descriptive to me, especially when I was starting. But it is true that `analysis` is very generic and  `staticAnalysisEngine` annoyingly long, `sbe` also does not sound right. I'll spend some time thinking about a different naming strategy. Then I'll either share the results here for further discussion or apply the steps you proposed in case I can't find a different solution.